File: oss-center/src/main/java/com/jmsoftware/maf/osscenter/write/entity/MergeResourceChunkPayload.java
package com.jmsoftware.maf.osscenter.write.entity;

import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import java.util.List;

/**
 * Description: MergeResourceChunkPayload, change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 8/12/2021 8:48 PM
 **/
@Data
public class MergeResourceChunkPayload {
    @NotBlank
    private String bucket;
    @NotEmpty
    private List<@Valid @NotBlank String> objectList;
}

File: oss-center/oss-center-infra/src/main/kotlin/com/jmsoftware/maf/osscenter/payload/MergeResourceChunkPayload.kt
package com.jmsoftware.maf.osscenter.payload

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty

/**
 * # MergeResourceChunkPayload
 *
 * Description: MergeResourceChunkPayload, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/17/22 8:34 AM
 */
class MergeResourceChunkPayload {
    @NotBlank
    lateinit var bucket: String
    @NotEmpty
    lateinit var objectList: List<String>
}
